show/main.py

```diff
@@ -768,23 +771,241 @@ def get_bgp_peer():
+            ns_str = str(ns)
+        else:
+            ns_str = " "
+        cmd = "sudo docker exec bgp{0} vtysh -c 'show ip route {1}'".format(ns_str, arg_strg)
+        output = commands.getoutput(cmd)
+        # in case no output or something went wrong with user specified cmd argument(s) error it out
+        # error from FRR always start with character "%"
+        if output == "":
+            return
+        if output[0] == "%":
+            # remove the "json" keyword that was added by this handler so it wont confuse the user
+            error_msg = output[:-4]
+            print error_msg
+            return
+        route_info = json.loads(output)
+        if filter_back_end:
+            # clean up the dictionary to remove all the nexthops that are back-end interface
+            for route, info in route_info.items():
+                new_info_l = []
+                new_info_cnt = 0
+                for i in range(0, len(info)):
+                    new_info = info[i]
+                    new_nhop_l = []
+                    del_cnt = 0
+                    while len(info[i]['nexthops']):
+                        nh = info[i]['nexthops'].pop(0)
+                        if "interfaceName" in nh:
+                            obj_type = constants.PORT_CHANNEL_OBJ
+                            if nh['interfaceName'].find('Ethernet') != -1:
+                                obj_type = constants.PORT_OBJ
+                            if device.skip_display(obj_type, nh['interfaceName']):
+                                del_cnt += 1
+                            else:
+                                new_nhop_l.append(nh)
+                        else:
+                            new_nhop_l.append(nh)
+                    # use the new filtered nhop list if it is not empty. if empty nexthop , this route is filtered out completely
+                    if len(new_nhop_l) > 0:
+                        new_info['nexthops'] = new_nhop_l
+                        new_info_cnt += 1
+                        # in case there are any nexthop that were deleted, we will need to adjust the nexhopt counts as well
+                        if del_cnt > 0:
+                            internalNextHopNum = info[i]['internalNextHopNum'] - del_cnt
+                            new_info['internalNextHopNum'] = internalNextHopNum
+                            internalNextHopActiveNum = info[i]['internalNextHopActiveNum'] - del_cnt
+                            new_info['internalNextHopActiveNum'] = internalNextHopActiveNum
+                        new_info_l.append(new_info)
+                if new_info_cnt:
+                    filtered_route[route] = new_info_l
+        else:
+            filtered_route = route_info
+        if not found_json:
+            #print out the header if this is not a json request
+            if not hdr_printed:
+                print_show_ip_route_hdr()
+                hdr_printed = True
+            print_ip_routes(filtered_route)
```

Comment:
all ASICs have the same route table, with some routes with different next hops. 

It looks like the routing table is being printed for each ASIC independently.  We should use prefix as key and merge next hops, and print a prefix only once.